### DISCUSSION TITLE
【官方】问题和建议收集

### DISCUSSION THREAD

**Discussion by apprat** [apprat/sober#18] [Ideas]
Sober 崇尚降低开发者的学习门槛，如果你对组件的一些api甚至命名有异议，可以在这里提出见解。

**Comment by Minemetero** [comment]
这几次更新写的更新日志。。。未免太匆忙了吧

**Reply by lingbopro** [reply]
附议，确实，对于一个稳定的项目来说还是对面MDUI更稳定

**Reply by apprat** [reply]
很抱歉对你造成困扰，精力都对文档进行了更新，小版本号的更新日志没有写上

**Comment by Minemetero** [comment]
增加新的功能我可以理解。。。但是为什么要删除原来的功能
![Screenshot 2024-11-21 091523](https://github.com/user-attachments/assets/3f0d5226-132f-4101-9561-faa0fb0f396a)

**Reply by apprat** [reply]
因为组件自身支持了输入框，所以不再需要嵌套input和textarea

**Comment by Minemetero** [comment]
最后一点。。。为什么npm的版本 ‘0.2.15’ 被删了

**Reply by apprat** [reply]
版本号命名出现了歧义且下载记录为0，可以切换到0.2.2

**Reply by kdxhub** [reply]
下载记录为0不代表被引用次数为0
下次使用SoberJs还是下载到本地再引用吧，毕竟MIT协议还是很方便的。

**Comment by lingbopro** [comment]
https://github.com/apprat/sober/commit/b131082c206e3798a1a2cdedc6b8abede676d37e#diff-9f0458604be8d94ab111a3d3df520bdd9417519ca4cb573f52e0374f3382e395

在这个提交里更改了默认主题颜色，但更新文档中并没有提及？

**Comment by xbenduan** [comment]
vue，react其实都有很成熟的UI库我觉得是不是给新兴的框架做适配功能更好（比如v-model）😏

**Reply by apprat** [reply]
可以例举一下，会逐渐开放支持。

**Reply by lingbopro** [reply]
v-model 已经适配了吧

**Reply by xbenduan** [reply]
> v-model 已经适配了吧

我的意思是那些新兴的框架也支持一些类似于v-model的功能，比如solid。给新兴的框架做好适配。这样这些框架的用户才会更多的使用这个UI库要不然像vue那些已经有很好的生态了很难有点去吸引那些开发者。但是solid的开发者就不一样了。solid还没有很流行的UI库

**Reply by apprat** [reply]
因为Solid使用的JSX ，所以对于sober也是支持的，目前缺少对Typescript的Solid类型声明，下个版本会补上

**Comment by Micropue** [comment]
why I can't use v-model on the s-text-field

**Reply by apprat** [reply]
v-model is supported in Sober v0.6 and above.

**Reply by Micropue** [reply]
![image](https://github.com/user-attachments/assets/ca0881bb-7d01-4da9-831b-532cd91fb8db)
it's my code's error?
it's doesn't work

**Reply by Micropue** [reply]
我用的是1.0.6

**Reply by apprat** [reply]
> 我用的是1.0.6

这里提供了一个可以正常运行的例子：

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8" />
  <title>Test</title>
</head>
<body>
  <main id="app">
    <s-text-field label="label" v-model="textRef"></s-text-field>
    {{textRef}}
  </main>
  <script src="https://unpkg.com/vue@3/dist/vue.global.js"></script>
  <script src="https://unpkg.com/sober@1.0.6/dist/sober.min.js"></script>
  <script>
    const { createApp, ref, reactive } = Vue

    const app = createApp({
      setup() {
        const textRef = ref('')
        return { textRef }
      }
    })
    app.config.compilerOptions.isCustomElement = (tag) => tag.startsWith('s-')
    app.mount('#app')
  </script>
</body>

</html>
````

**Reply by apprat** [reply]
> 我用的是1.0.6

If you still can't bind, check the console to see if it's throwing an error?

**Reply by Micropue** [reply]
那为啥我这么写不好使呀，我用插槽的话是会影响这个吗

**Reply by Micropue** [reply]
没有任何报错，但是输出出来的username或者是password就是空的。
![image](https://github.com/user-attachments/assets/5e2c348c-4292-4ab9-b8d9-d28ab5714a32)

**Reply by apprat** [reply]
> 那为啥我这么写不好使呀，我用插槽的话是会影响这个吗

可以监听一下text-field的@input事件是否正常执行？

**Reply by Micropue** [reply]
正常的

**Reply by apprat** [reply]
> 正常的

我没法推测原因，可能是你的构建工具引入了比较旧的包缓存，可以尝试重新安装 sober

**Reply by Micropue** [reply]
可以加个qq吗2857903623你看一下，我之前是纯html css js刚接触vue

**Reply by apprat** [reply]
你可以添加一下官方交流群

**Reply by Micropue** [reply]
在哪里

**Comment by fantasy704** [comment]
按照文档说明使用text field或者checkbox的v-model绑定变量都没有生效，怎么解决？

**Reply by apprat** [reply]
> 按照文档说明使用text field或者checkbox的v-model绑定变量都没有生效，怎么解决？

https://github.com/apprat/sober/issues/34#issuecomment-2628860600

**Comment by Minemetero** [comment]
<img width="961" alt="{D261A225-B594-48FF-BC89-F4EE22BBA648}" src="https://github.com/user-attachments/assets/e9bf22f4-e62b-4347-8fe7-b766bced2125" />
这种写死在代码里的东西。。不太好

**Reply by apprat** [reply]
> <img alt="{D261A225-B594-48FF-BC89-F4EE22BBA648}" width="961" src="https://private-user-images.githubusercontent.com/156386015/444956653-e9bf22f4-e62b-4347-8fe7-b766bced2125.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc2MjQ0NjksIm5iZiI6MTc0NzYyNDE2OSwicGF0aCI6Ii8xNTYzODYwMTUvNDQ0OTU2NjUzLWU5YmYyMmY0LWU2MmItNDM0Ny04ZmU3LWI3NjZiY2VkMjEyNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUxOVQwMzA5MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDk0ZGFiYTc3ZjM3ZGQ4ZjJhOGJhNzkxOWI2NzE1NTI0YWViOTc0NTI4MGQyZGM0ZTAzODQ4OWVlYzk3ODU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ukqdEwaI54CDYnkhtcRJnPgcRiSmi07PNSyCkcgvxYQ"> 这种写死在代码里的东西。。不太好

默认只支持中英文，未来也不会再默认支持其他语言，如果需要支持其他语言，需要自己调用 addLocale 添加

**Comment by kdxhub** [comment]
为什么文档里面的配色生成工具没了
这边看到的是空白的，只有介绍文字
https://soberjs.com/style/theme-builder

**Reply by kdxhub** [reply]
![Screenshot_2025-06-26-23-08-03-147_com microsoft emmx](https://github.com/user-attachments/assets/af704ec5-8591-4076-b27a-3c24c606a26e)

**Reply by Minemetero** [reply]
lol. Very stable.

**Reply by lingbopro** [reply]
这sober可太《稳定》了🤣

**Reply by Minemetero** [reply]
> 这sober可太《稳定》了🤣

你做你的烂尾工程吧

**Reply by lingbopro** [reply]
> > 这sober可太《稳定》了🤣
> 
> 你做你的烂尾工程吧

都说了没烂尾，在重构（）

**Reply by Minemetero** [reply]
哎呀这sober已经一个月没更新了啊

**Reply by lingbopro** [reply]
> 哎呀这sober已经一个月没更新了啊

sober是这样的

**Reply by Minemetero** [reply]
> > 哎呀这sober已经一个月没更新了啊
> 
> sober是这样的

那个Alpha Branch更新了

**Reply by lingbopro** [reply]
> > > 哎呀这sober已经一个月没更新了啊
> > 
> > sober是这样的
> 
> 那个Alpha Branch更新了

那很好了

**Comment by Minemetero** [comment]
@lingbopro I have a dream.
我要把官方文档全部英文翻译一遍

**Reply by lingbopro** [reply]
好啊，很好啊

**Reply by Minemetero** [reply]
说句我个人的主观感受，Material 3好像也没啥变化啊

**Comment by Minemetero** [comment]
1.1.9 的更新日志怎么被吃了 😭

**Reply by lingbopro** [reply]
sober日常吃更新日志

**Comment by LiMao00** [comment]
如果我有在 html 里添加 `font-size: 40px` 的方式放大界面的需求，组件是怎么做适配的？我看到的是组件内文本是 rem 单位所以跟随放大了，但是边距、布局等就错乱了

**Reply by lingbopro** [reply]
sober的布局里大量使用绝对尺寸，所以布局乱也正常（）

**Reply by apprat** [reply]
严格来说这是个设计缺陷：

当初使用rem作为字体单位的目的是rem会根据系统设置的字体大小来按比例调整，但如果你使用rem作为布局单位，这就违法了sober的设计，目前去解决这个缺陷没有太多意义。

v2版本会解决这个问题，让文字的根大小不再依赖rem，改为依赖 `s-page` 组件的CSS变量去调整比例。

**Reply by apprat** [reply]
如果你只是在移动端希望获得“更大的界面”，可以考虑使用
```html
<meta name="viewport" content="width=device-width,initial-scale=0.8,minimum-scale=0.8,maximum-scale=0.8,user-scalable=no" />
````
来调整整体比例。

**Comment by fantasy704** [comment]
Dialog对话框在按下Esc后关闭，但是第二次再次点击trigger后打不开，底部面板组件也有这个问题。

**Reply by Micropue** [reply]
这个是因为它用的原生Dialog改的样式，这样可以省去处理zIndex ，但是没有取消默认事件就是ESC会导致原生Dialog关闭。
